test/itkAttenuationImageFilterTest.cxx

```diff
@@ -50,14 +50,42 @@ itkAttenuationImageFilterTest(int argc, char * argv[])
   using AttenuationFilterType = itk::AttenuationImageFilter<SpectraImageType, OutputImageType, MaskImageType>;
   AttenuationFilterType::Pointer attenuationFilter = AttenuationFilterType::New();
 
+  // Verify spatial distances are not settable without input image for reference
+  ITK_TRY_EXPECT_EXCEPTION(attenuationFilter->SetFixedEstimationDepthMM(3.0));
+  ITK_TRY_EXPECT_EXCEPTION(attenuationFilter->SetPadUpperBoundsMM(3.0));
+  ITK_TRY_EXPECT_EXCEPTION(attenuationFilter->SetPadLowerBoundsMM(3.0));
+
   attenuationFilter->SetInput(inputImage);
   ITK_TEST_SET_GET_VALUE(inputImage, attenuationFilter->GetInput());
 
+  // Verify filter does not run without mask
+  ITK_TRY_EXPECT_EXCEPTION(attenuationFilter->Update());
+
   attenuationFilter->SetMaskImage(maskImage);
   ITK_TEST_SET_GET_VALUE(maskImage, attenuationFilter->GetMaskImage());
 
+  // Bad scanline direction produces an error
+  attenuationFilter->SetScanDirection(Dimension);
+  ITK_TRY_EXPECT_EXCEPTION(attenuationFilter->Update());
+
+  attenuationFilter->SetScanDirection(0);
+  ITK_TEST_SET_GET_VALUE(0, attenuationFilter->GetScanDirection());
+
+  // Missing sampling frequency produces an error
+  ITK_TRY_EXPECT_EXCEPTION(attenuationFilter->Update());
+
   attenuationFilter->SetSamplingFrequencyMHz(60);
   ITK_TEST_SET_GET_VALUE(60.0, attenuationFilter->GetSamplingFrequencyMHz());
+  ITK_TRY_EXPECT_NO_EXCEPTION(attenuationFilter->Update());
+
+  attenuationFilter->SetFixedEstimationDepth(3);
```

Comment:
Should this be FixedEstimationDepth**MM**? Or maybe pass a more realistic value here, like 20 pixels.

Reply:
This line serves to test setting a pixel distance before we test setting a user-specified spatial distance on line 86. The test input image appears to have a scanline spacing of 0.6mm, so a spatial distance of 3.0mm becomes approximately 5 pixels, which nearly matches the contrived value of 3 pixels used here.

Does that match your expectations or are there concerns about the test data?

Reply:
> test setting a pixel distance

You are right. It is somewhat realistic, and for test any value will do.